apis/pipelines/v1alpha5/provider_conversion.go

```diff
@@ -12,6 +15,42 @@ func (src *Provider) ConvertTo(dstRaw conversion.Hub) error {
 	dst := dstRaw.(*hub.Provider)
 	dstApiVersion := dst.APIVersion
 
+	var beamArgsPatchOps []common.JsonPatchOperation
+
+	for _, nv := range src.Spec.DefaultBeamArgs {
+		patchOp := common.JsonPatchOperation{
+			Op:   "add",
+			Path: "/framework/parameters/beamArgs/0",
+			Value: map[string]string{
+				"name":  nv.Name,
+				"value": nv.Value,
+			},
+		}
+
+		beamArgsPatchOps = append(beamArgsPatchOps, patchOp)
+	}
```

Comment:
```suggestion
	for _, nameValue := range src.Spec.DefaultBeamArgs {
		patchOp := common.JsonPatchOperation{
			Op:   "add",
			Path: "/framework/parameters/beamArgs/0",
			Value: map[string]string{
				"name":  nameValue.Name,
				"value": nameValue.Value,
			},
		}

		beamArgsPatchOps = append(beamArgsPatchOps, patchOp)
	}
```

starting to think this nameValue could have a helper method on the stuct that just does a conversion to the map[string]string{} type

Reply:
apart from this conversion code does it happen anywhere else?

Reply:
doubtful....so yeah probs just easier to change and ignore the helper func